pkg/types/clusterconfig/network_validations.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2021 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterconfig
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/servicequotas"
+	"github.com/cortexlabs/cortex/pkg/lib/aws"
+	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	"github.com/cortexlabs/cortex/pkg/lib/pointer"
+	"github.com/cortexlabs/cortex/pkg/lib/sets/strset"
+)
+
+const (
+	_elasticIPsQuotaCode         = "L-0263D0A3"
+	_internetGatewayQuotaCode    = "L-A4707A72"
+	_natGatewayQuotaCode         = "L-FE5A380F"
+	_vpcQuotaCode                = "L-F678F1CE"
+	_securityGroupsQuotaCode     = "L-E79EC296"
+	_securityGroupRulesQuotaCode = "L-0EA8095F"
+)
+
+func VerifyNetworkQuotas(
+	awsClient *aws.Client,
+	requiredInternetGateways int,
+	natGatewayRequired bool,
+	highlyAvailableNATGateway bool,
+	requiredVPCs int,
+	availabilityZones strset.Set,
+	numNodeGroups int,
+	longestCIDRWhiteList int) error {
+
+	desiredQuotaCodes := strset.New(_elasticIPsQuotaCode, _internetGatewayQuotaCode, _natGatewayQuotaCode, _vpcQuotaCode, _securityGroupsQuotaCode, _securityGroupRulesQuotaCode)
+	quotaCodeToValueMap := map[string]int{}
+
+	err := awsClient.ServiceQuotas().ListServiceQuotasPages(
+		&servicequotas.ListServiceQuotasInput{
+			ServiceCode: pointer.String("ec2"),
+		},
+		func(page *servicequotas.ListServiceQuotasOutput, lastPage bool) bool {
+			if page == nil {
+				return false
+			}
+			for _, quota := range page.Quotas {
+				if quota == nil || quota.QuotaCode == nil || quota.Value == nil {
+					continue
+				}
+				if desiredQuotaCodes.Has(*quota.QuotaCode) {
+					quotaCodeToValueMap[*quota.QuotaCode] = int(*quota.Value)
+					return false
+				}
+			}
+			return true
+		},
+	)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	err = awsClient.ServiceQuotas().ListServiceQuotasPages(
+		&servicequotas.ListServiceQuotasInput{
+			ServiceCode: pointer.String("vpc"),
+		},
+		func(page *servicequotas.ListServiceQuotasOutput, lastPage bool) bool {
+			if page == nil {
+				return false
+			}
+			for _, quota := range page.Quotas {
+				if quota == nil || quota.QuotaCode == nil || quota.Value == nil {
+					continue
+				}
+				if desiredQuotaCodes.Has(*quota.QuotaCode) {
+					quotaCodeToValueMap[*quota.QuotaCode] = int(*quota.Value)
+				}
+			}
+			return true
+		},
+	)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	skippedValidations := ""
```

Comment:
nit: consider `var skippedValidations []string`, and then joining by `\n` in the deferred function